packages/mobile/src/home/WalletHome.tsx

```diff
@@ -1,212 +1,156 @@
-  onScroll: (event: NativeSyntheticEvent<NativeScrollEvent>) => void
-
-  constructor(props: Props) {
-    super(props)
+function WalletHome() {
+  const { t } = useTranslation(Namespaces.walletFlow5)
+
+  const appState = useSelector(appStateSelector)
+  const isLoading = useSelector((state) => state.home.loading)
+  const recipientCache = useSelector(phoneRecipientCacheSelector)
+  const showTokensInHome = useSelector(multiTokenShowHomeBalancesSelector)
+  const balances = useSelector(balancesSelector)
+  const cashInButtonExpEnabled = useSelector((state) => state.app.cashInButtonExpEnabled)
+
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+
+  const dispatch = useDispatch()
+
+  const showTestnetBanner = () => {
+    dispatch(
+      showMessage(
+        t('testnetAlert.1', { testnet: _.startCase(DEFAULT_TESTNET) }),
+        ALERT_BANNER_DURATION,
+        null,
+        null,
+        t('testnetAlert.0', { testnet: _.startCase(DEFAULT_TESTNET) })
+      )
+    )
+  }
 
-    this.scrollPosition = new Animated.Value(0)
-    this.onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: this.scrollPosition } } }])
-    this.state = {
-      isMigrating: false,
+  const tryImportContacts = async () => {
+    // Skip if contacts have already been imported
+    if (Object.keys(recipientCache).length) {
+      return
     }
-  }
 
-  onRefresh = async () => {
-    this.props.refreshAllBalances()
+    const hasGivenContactPermission = await checkContactsPermission()
+    if (hasGivenContactPermission) {
+      dispatch(importContacts())
+    }
   }
 
-  componentDidMount = () => {
+  useEffect(() => {
     // TODO find a better home for this, its unrelated to wallet home
-    this.props.initializeSentryUserContext()
+    dispatch(initializeSentryUserContext())
     if (SHOW_TESTNET_BANNER) {
-      this.showTestnetBanner()
+      showTestnetBanner()
     }
 
-    // TODO: Fire refreshAllBalances when the app state changes to active. It's easier to do that when we
-    // transform this into a function component.
-    // useEffect(() => {
-    //   if (appState === AppState.Active) {
-    //     dispatch(refreshAllBalances())
-    //   }
-    // }, [appState])
-    this.props.refreshAllBalances()
     // Waiting 1/2 sec before triggering to allow
     // rest of feed to load unencumbered
-    setTimeout(this.tryImportContacts, 500)
-  }
+    setTimeout(tryImportContacts, 500)
+  }, [])
 
-  shouldShowCashInBottomSheet = () => {
-    const hasStable = STABLE_CURRENCIES.some((currency) =>
-      this.props.balances[currency]?.isGreaterThan(STABLE_TRANSACTION_MIN_AMOUNT)
-    )
-    const hasGold = this.props.balances[Currency.Celo]?.isGreaterThan(GOLD_TRANSACTION_MIN_AMOUNT)
-    const isAccountBalanceZero = !hasStable && !hasGold
+  useEffect(() => {
+    if (appState === AppState.Active) {
+      dispatch(refreshAllBalances())
```

Comment:
not directly related to this PR, but I noticed due to the way the sagas are initialized, this action isn't handled by any saga (still waiting in the blocking `appInit` saga).
But maybe we can remove this soon anyway with the multi token changes?

Reply:
We still need to fetch balances on app load, for now I just added `yield call(refreshBalances)` to the saga initialization, but we should think about how to handle the token fetching in a more robust way. I think a good time to do that is when we remove the old code